src/mqt/bench/benchmarks/qiskit_application_optimization/routing.py

```diff
@@ -81,28 +91,27 @@ def binary_representation(self, x_sol=0):
         c = 2 * A * (n - 1) + 2 * A * (K**2)
 
         try:
-            max(x_sol)
-
             # Evaluates the cost distance from a binary representation of a path
-            def fun(x):
-                return np.dot(np.around(x), np.dot(Q, np.around(x))) + np.dot(g, np.around(x)) + c
+            def fun(x: NDArray[np.float_]) -> float:
+                return cast(float, np.dot(np.around(x), np.dot(Q, np.around(x))) + np.dot(g, np.around(x)) + c)
 
             cost = fun(x_sol)
         except Exception:
             cost = 0
 
-        return Q, g, c, cost
+        return cast(NDArray[np.float_], Q), cast(NDArray[np.float_], g), cast(float, c), cost
```

Comment:
are the first two casts really necessary here?

Reply:
Good catch, it was possible to remove those.